### PR TITLE
promote Bigqueryconnection to v1beta1 API

### PR DESCRIPTION
**Pull request by 600lyy**
### Change description
Promote bigqueryconnection API to v1beta1

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by 600lyy** [comment]
@jingyih pls review

File: pkg/test/resourcefixture/testdata/basic/bigqueryconnection/v1alpha1/bigqueryconnectionconnection/cloudsqlconnectionbasic/create.yaml [hunk 1]
```diff
@@ -24,7 +24,8 @@ spec:
     instanceRef:
       # external: "projects/${projectId}/locations/us-central1/instances/sqlinstance-sample-001"
       name: sqlinstance-sample-${uniqueId}
-    database: sqldatabase-sample-${uniqueId}
+    databaseRef: 
+      name: sqldatabase-sample-${uniqueId}
```
Comment (jingyih) [inline pkg/test/resourcefixture/testdata/basic/bigqueryconnection/v1alpha1/bigqueryconnectionconnection/cloudsqlconnectionbasic/create.yaml:28]: Cool!

File: pkg/test/resourcefixture/testdata/basic/bigqueryconnection/v1alpha1/bigqueryconnectionconnection/cloudsqlconnectionbasic/_http.log [hunk 2]
```diff
@@ -1188,7 +1188,7 @@ X-Xss-Protection: 0
 
 {
   "cloudSql": {
-    "database": "sqldatabase-sample-${uniqueId}",
+    "database": "projects/${projectId}/instances/sqlinstance-sample-${uniqueId}/databases/sqldatabase-sample-${uniqueId}",
```
Comment (jingyih) [inline pkg/test/resourcefixture/testdata/basic/bigqueryconnection/v1alpha1/bigqueryconnectionconnection/cloudsqlconnectionbasic/_http.log:1191]: This is unexpected. I don't think this works with real GCP API? My understanding is that the underlying GCP API expects the name of the database.

File: pkg/controller/direct/bigqueryconnection/connection_controller.go [hunk 3]
```diff
@@ -106,6 +106,13 @@ func (a *Adapter) normalizeReference(ctx context.Context) error {
 			}
 			sql.InstanceRef.External = instance.ConnectionName()
 		}
+		if sql.DatabaseRef != nil {
+			database, err := refs.ResolveSQLDatabaseRef(ctx, a.reader, obj, sql.DatabaseRef)
+			if err != nil {
+				return err
+			}
+			sql.DatabaseRef.External = database.String()
```
Comment (jingyih) [inline pkg/controller/direct/bigqueryconnection/connection_controller.go:114]: I think we only need database name for this field.

File: apis/refs/v1beta1/sqldatabaseref.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	}
+
+	if ref.Name == "" && ref.External == "" {
+		return nil, fmt.Errorf("must specify either name or external on databaseRef")
+	}
+	if ref.External != "" && ref.Name != "" {
+		return nil, fmt.Errorf("cannot specify both spec.databaseRef.name and spec.databaseRef.external")
+	}
+
+	if ref.External != "" {
```
Comment (jingyih) [inline apis/refs/v1beta1/sqldatabaseref.go:60]: Users should be able to provide database name directly via the `external` field. e.g.

```
databaseRef:
  external: sqldatabase-sample-${uniqueId}
```

File: apis/refs/v1beta1/sqldatabaseref.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		Kind:    "SQLDatabase",
+	})
+	if err := reader.Get(ctx, key, sqldatabase); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("referenced SQL databse %v not found", key)
+		}
+		return nil, fmt.Errorf("error reading referenced SQL databse %v: %w", key, err)
+	}
+
+	resourceID, _, err := unstructured.NestedString(sqldatabase.Object, "spec", "resourceID")
```
Comment (jingyih) [inline apis/refs/v1beta1/sqldatabaseref.go:95]: It seems to be that the resourceID is the name of the database? e.g. https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/790c54b1bae78e405615273abbd7e54376180a7d/pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqldatabase/_generated_object_sqldatabase.golden.yaml#L23

So we just need to return resourceID? (At least for the datasetRef in BigQueryConnectionConnectoin, the GCP API does not need project ID or instance ID of the dataset)

File: pkg/test/resourcefixture/testdata/basic/bigqueryconnection/v1alpha1/bigqueryconnectionconnection/cloudsqlconnectionbasic/_http.log (hunk 2 above)
Comment (600lyy) [reply pkg/test/resourcefixture/testdata/basic/bigqueryconnection/v1alpha1/bigqueryconnectionconnection/cloudsqlconnectionbasic/_http.log:1191]: According to the API doc,  this field is just the name of db

**Review by jingyih — COMMENTED** [review]
The only blocking comment is that we need to update the alpha API and regenerate the CRD. Other comments can be addressed in a followup PR.

**Comment by jingyih** [comment]
@maqiuyujoyce @yuwenma How do we handle the URI skeleton test when we migrate a resource to direct? Currently the unit test panics.

https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/e12f6d0a871173fb2244d06da1e8045aedd157b0/pkg/resourceskeleton/testdata/uri-skeleton.yaml#L765-L767

File: config/samples/resources/bigqueryconnectionconnection/cloudspanner-connection/bigqueryconnection_v1beta1_bigqueryconnectionconnection.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+metadata:
+  name: bigqueryconnectionconnection-sample
+spec:
+  location: us-central1
+  projectRef:
+    # Replace ${PROJECT_ID?} with your project ID
+    external: ${PROJECT_ID?}
+  cloudSpanner:
+    databaseRef:
+      name: bigqueryconnection-dep
```
[the remark this answers is not in the document]
Comment (600lyy) [reply config/samples/resources/bigqueryconnectionconnection/cloudspanner-connection/bigqueryconnection_v1beta1_bigqueryconnectionconnection.yaml:26]: According to the document, the length of name cannot exceed max 15 characters

File: config/samples/resources/bigqueryconnectionconnection/spark-connection/bigqueryconnection_v1beta1_bigqueryconnectionconnection.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+apiVersion: bigqueryconnection.cnrm.cloud.google.com/v1beta1
+kind: BigQueryConnectionConnection
+metadata:
+  name: bigqueryconnectionconnection-sample
+spec:
+  location: us-central1
+  projectRef:
+    # Replace ${PROJECT_ID?} with your project ID
+    external: ${PROJECT_ID?}
+  spark: {}
```
[the remark this answers is not in the document]
Comment (600lyy) [reply config/samples/resources/bigqueryconnectionconnection/spark-connection/bigqueryconnection_v1beta1_bigqueryconnectionconnection.yaml:24]: Currently, Mockgcp service for Dataproc has not been implemented. Thus I suggest we add this in a followup PR

**Pull request merged**